docs/source/advanced_settings.rst
Advanced Settings
=================

Conf files
----------
MozDef python scripts in almost all cases expect to be given a -c path/to/file.conf command line option to specify configuration/run time options.

These files all follow the same format:

::

    [options]
    setting1=value1
    setting2=value2


All programs do their best to set reasonable, sane defaults and most will run fine without a conf file. By default programname.py will look for programname.conf as it's configuration file so if you follow that convention you don't even need to specify the -c path/to/file.conf option.

You can override any .conf file setting by providing an equivalent environment variable.

Certain environment variables have special meaning to the web ui (meteor):

::

    OPTIONS_METEOR_ROOTURL is "http://localhost" by default and should be set to the dns name of the UI where you will run MozDef
    OPTIONS_METEOR_PORT is 80 by default and is the port on which the UI will run
    OPTIONS_METEOR_ROOTAPI is http://rest:8081 by default and should resolve to the location of the rest api
    OPTIONS_KIBANAURL is http://localhost:9090/app/kibana# by default and should resolve to your kibana installation
    OPTIONS_METEOR_ENABLECLIENTACCOUNTCREATION is true by default and governs whether accounts can be created
    OPTIONS_METEOR_AUTHENTICATIONTYPE is meteor-password by default and can be set to oidc to allow for oidc authentication
    OPTIONS_REMOVE_FEATURES is empty by default, but if you pass a comma separated list of features you'd like to remove they will no longer be available.

You can find a list of features in the meteor/private/features.txt file in the git repo.


Special Config Items
_____________________

Here are some tips for some key settings:

::

    [options]
    esservers=http://server1:9200,http://server2:9200,http://server3:9200

is how you can specify servers in your elastic search cluster.

::

    [options]
    backup_indices = intelligence,.kibana,alerts,events,complianceitems,.jsp,.marvel-kibana,vulnerabilities
    backup_dobackup = 1,1,1,1,1,1,1,1
    backup_rotation = none,none,monthly,daily,none,none,none,none
    backup_pruning = 0,0,0,20,0,0,0,0

is how you would configure the backupSnapshot.py and pruneIndexes.py programs to backup selected elastic search indexes, rotate selected indexes and prune certain indexes at selected intervals. In the case above we are backing up all indexes mentioned, rotating alerts monthly, rotating events daily and pruning events indices after 20 days.

::

    [options]
    autocategorize = True
    categorymapping = [{"bruteforce":"bruteforcer"},{"nothing":"nothing"}]

is how you would configure collectAttackers.py to do autocategoization of attackers that it discovers and specify a list of mappings matching alert categories to attacker category.


CHANGELOG
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

## [Unreleased]

## [v1.38.5] - 2019-04-09
### Added
- Support for CSS themes

### Changed
- The CI/CD order to now build docker images in CodeBuild, upload them
  to DockerHub and then pull them down in the packer instance. Updated docs.
- Assert TravisCI Python version in advance of change of Travis default to 3.6

### Fixed
- Dashboard error on docker spinup


## [v1.38.4] - 2019-04-08
### Fixed
- Docker image tagging for git version tag builds
- Correctly propagate the source ip address to the details.sourceipaddress in Duo logpull
- Invalid literal in squidFixup.py destionationport field
- Lowercase TAGS in squidFixup.py
- Adding check for None type object in date fields to address GuardDuty null date

### Added
- Documentation on the CI/CD process
- A summary to squidFixup.py
- Tags assertions to tests

## [v1.38.3] - 2019-04-01
### Fixed
- AWS CodeBuild tag semver regex

## [v1.38.2] - 2019-03-29
### Fixed
- Remaining references to old alertplugins container

## [v1.38.1] - 2019-03-29
### Added
- Enable CI/CD with AWS CodeBuild
- Create AMIs of MozDef, replicate and share them
- Link everything (container images, AMIs, templates) together by MozDef version

### Changed
- Publish versioned CloudFormation templates
- RabbitMQ configured to use a real password

## [v1.38] - 2019-03-28
### Added
- Create alert plugins with ability to modify alerts in pipeline

### Changed
- Renamed existing alertplugin service to alertactions
- Updated rabbitmq docker container to 3.7

### Fixed
- Resolved sshd mq plugin to handle more types of events

## [v1.37] - 2019-03-01
### Added
- Watchlist - use the UI to quickly add a term (username, IP, command, etc.) that MozDef alerts on
- Generic Deadman - use a simple config file to validate that expected events are appearing in a given time window (and alert an Error when they do not)

### Changed
- Improve error handling on Slack bot
- Improve Slack bot alert format for better readability
- Minor UI adjustments

### Fixed
- Some Duo events were not correctly displaying the source IP address. It is now always the access device IP
- Fixed defaults for Slack bot to ensure more consistency each time it loads
- Added checks on sending SQS messages to only accept intra-account messages
- Improved docker performance and disk space requirements

[Unreleased]: https://github.com/mozilla/MozDef/compare/v1.38.5...HEAD
[v1.38.5]: https://github.com/mozilla/MozDef/compare/v1.38.4...v1.38.5
[v1.38.4]: https://github.com/mozilla/MozDef/compare/v1.38.3...v1.38.4
[v1.38.3]: https://github.com/mozilla/MozDef/compare/v1.38.2...v1.38.3
[v1.38.2]: https://github.com/mozilla/MozDef/compare/v1.38.1...v1.38.2
[v1.38.1]: https://github.com/mozilla/MozDef/compare/v1.38...v1.38.1
[v1.38]: https://github.com/mozilla/MozDef/compare/v1.37...v1.38
[v1.37]: https://github.com/mozilla/MozDef/releases/tag/v1.37


meteor/imports/settings.js
/*
This Source Code Form is subject to the terms of the Mozilla Public
License, v. 2.0. If a copy of the MPL was not distributed with this
file, You can obtain one at http://mozilla.org/MPL/2.0/.
Copyright (c) 2017 Mozilla Corporation
*/

// declare truthy sanity
var trueValues = ['1', 1, 'true', true, 'yes','yup','certainly','always'];
var falseValues = ['0', 0, 'false', false, 'no', undefined, null,'nope','never'];

//configuration settings
if (Meteor.isServer) {
    // Figure out truthiness of the desire to enable Client Account Creation.
    // This allows for env override of default true value
    // with anything that is in the falseValues list above.
    // If env variable isn't set, it's undefined and will || true as the default
    var allowAccountCreation= new Boolean(!falseValues.includes( (process.env.OPTIONS_METEOR_ENABLECLIENTACCOUNTCREATION || true ))).valueOf()
    mozdef = {
        rootURL: process.env.OPTIONS_METEOR_ROOTURL || "http://localhost",
        port: process.env.OPTIONS_METEOR_PORT || "80",
        rootAPI: process.env.OPTIONS_METEOR_ROOTAPI || "http://rest:8081",
        kibanaURL: process.env.OPTIONS_METEOR_KIBANAURL || "http://localhost:9090/app/kibana",
        enableClientAccountCreation: allowAccountCreation,
        authenticationType: process.env.OPTIONS_METEOR_AUTHENTICATIONTYPE || "meteor-password",
        removeFeatures: process.env.OPTIONS_REMOVE_FEATURES || ""
    }

    // send these settings to the client via the Meteor.settings.public
    // reactive object
    // Note that:
    // Meteor.settings.public=mozdef;
    // doesn't work as you can't override the root 'public'
    // but you can set public.mozdef, so we do that.
    Meteor.settings.public.mozdef=mozdef;
    console.log(Meteor.settings);

}

if (Meteor.isClient) {
    mozdef=Meteor.settings.public.mozdef;
}